### DISCUSSION TITLE
WikiPDFExport: blank/empty pdf output file

### DISCUSSION THREAD

**Discussion by COWI-FIBR** [rfennell/AzurePipelines#1196] [Q&A] [answered]
In the pipeline I start by generating the wiki page, adding it to the wiki, and in the the final step export it to pdf.
I.e., I use the already created *.md file that has been added to the wiki and is now available in the System.DefaultWorkingDirectory.

PdfExport task executes fine, but the generated pdf is blank. Opening the file, the page is white, nothing else.

From the log of executing WikiPdfExportTask@2 step:
> A filename '**20220126.3_project_releasenote.md**' in the folder '**/home/vsts/work/1/s**' has been requested so only processing that file 
> Adding the verbose flag to increase logging
> **Changing folder to /home/vsts/work/1/s**
> Using command '"/home/vsts/work/_temp/azuredevops-export-wiki"  **-p "/home/vsts/work/1/s"** **-s "20220126.3_project_releasenote.md"** **-o "/home/vsts/work/1/s/20220126.3_project_releasenote.pdf"** -v'
> Reading .order file in directory /home/vsts/work/1/s
> Converting Markdown to HTML
> No CSS specified, using devopswikistyle.css
> Converting HTML to PDF
> PDF created at: **/home/vsts/work/1/s/20220126.3_project_releasenote.pdf**

I've added a Bash step to `cat` the 20220126.3_project_releasenote.md file (although I can also see the correctly generated release note on the wiki) and list the content `ls` of System.DefaultWorkingDirectory to verify that both files are there.

> 4.0K -rw-r--r-- 1 vsts docker 1022 Jan 26 14:14 20220126.3_project_releasenote.md
> 4.0K -rw-r--r-- 1 vsts docker  857 Jan 26 14:14 20220126.3_project_releasenote.pdf

Anyone else experienced that the output from pdf exporter is blank?

The PdfExport task:
>   task: WikiPdfExportTask@2
>   displayName: 'Wiki to pdf - PAT'
>   inputs:
>     cloneRepo: false
>     singleFile: '$(Build.BuildNumber)_project_releasenote.md'
>     user: 'xxx'
>     password: '$(PAT)'
>     localpath: '$(System.DefaultWorkingDirectory)'
>     outputFile: '$(System.DefaultWorkingDirectory)/$(Build.BuildNumber)_project_releasenote.pdf'

**Comment by rfennell** [comment, marked as answer]
This task is a thin wrapper around [MaxMelcher's AzureDevOps.WikiPDFExport command line tool.](https://github.com/MaxMelcher/AzureDevOps.WikiPDFExport). My task log shows the command line it is running so you can experiment outside an Azure DevOps Pipeline.

I don't know anything about the internal workings of Max's tools, so I would suggest you ask for help on that project's GitHub repo.

**Reply by COWI-FIBR** [reply]
Thank you, I'll post the question over there.
